components/DashboardProfile/index.js

```diff
@@ -29,34 +36,54 @@ class DashboardProfile extends React.Component {
   handleEdit = () => {
     this.setState({ edit: true });
   };
+  validateEmail(email) {
+    var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(email);
+  }
   handleChangeUserMetaData = (field, e) => {
-    let value;
-    if (!e.target.value) {
-      value = this.props.profile.nickname;
-    } else {
-      value = e.target.value;
+    switch (field) {
+      case 'userEmail':
+        if (this.validateEmail(e.target.value)) {
+          this.setState({
+            disabled: false,
+            user_metadata: {
+              ...this.state.user_metadata,
+              userEmail: e.target.value
+            }
+          });
+        } else {
+          if (e.target.value) {
+            this.setState({ disabled: true });
+            swal(this.messages.emailValidation, null, 'error');
+          } else {
+            this.setState({ disabled: false });
+          }
+        }
+      default:
+        this.setState({
+          user_metadata: {
+            ...this.state.user_metadata,
+            [field]: e.target.value
+          }
+        });
```

Comment:
is it possible to remove this duplication of:
```
          this.setState({
            disabled: false,
            user_metadata: {
              ...this.state.user_metadata,
              userEmail: e.target.value
            }
          });
``` 

perhaps refactor the first case to `if(!this.validateEmail)`? 

And use the fact that a switch statement will "fall through" unless you use a `break` [here](https://stackoverflow.com/questions/13207927/switch-statement-multiple-cases-in-javascript#13207939)

Reply:
check